test/Microsoft.Health.Fhir.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs

```diff
@@ -64,30 +92,52 @@ public CosmosDbFhirStorageTestsFixture()
             var upgradeManager = new FhirCollectionUpgradeManager(updaters, _cosmosDataStoreConfiguration, optionsMonitor, dbLock, NullLogger<FhirCollectionUpgradeManager>.Instance);
             IDocumentClientTestProvider testProvider = new DocumentClientReadWriteTestProvider();
 
-            var documentClientInitializer = new DocumentClientInitializer(testProvider, NullLogger<DocumentClientInitializer>.Instance);
+            var fhirRequestContextAccessor = new FhirRequestContextAccessor();
+
+            var documentClientInitializer = new FhirDocumentClientInitializer(testProvider, fhirRequestContextAccessor, NullLogger<FhirDocumentClientInitializer>.Instance);
             _documentClient = documentClientInitializer.CreateDocumentClient(_cosmosDataStoreConfiguration);
             var fhirCollectionInitializer = new CollectionInitializer(_cosmosCollectionConfiguration.CollectionId, _cosmosDataStoreConfiguration, _cosmosCollectionConfiguration.InitialCollectionThroughput, upgradeManager, NullLogger<CollectionInitializer>.Instance);
-            documentClientInitializer.InitializeDataStore(_documentClient, _cosmosDataStoreConfiguration, new List<ICollectionInitializer> { fhirCollectionInitializer }).GetAwaiter().GetResult();
+
+            // Cosmos DB emulators throws errors when multiple collections are initialized concurrently.
+            // Use the semaphore to only allow one initialization at a time.
+            await CollectionInitializationSemaphore.WaitAsync();
+
+            try
+            {
+                await documentClientInitializer.InitializeDataStore(_documentClient, _cosmosDataStoreConfiguration, new List<ICollectionInitializer> { fhirCollectionInitializer });
+            }
+            finally
+            {
+                CollectionInitializationSemaphore.Release();
+            }
 
             var cosmosDocumentQueryFactory = new FhirCosmosDocumentQueryFactory(Substitute.For<IFhirRequestContextAccessor>(), NullFhirDocumentQueryLogger.Instance);
-            var fhirRequestContextAccessor = new FhirRequestContextAccessor();
 
-            Value = new CosmosFhirDataStore(
-                new NonDisposingScope(_documentClient),
+            var documentClient = new NonDisposingScope(_documentClient);
```

Comment:
Ideally we would dispose this

Reply:
This is disposed in Dispose method above (that was new coded added).